iina/HistoryController.swift

```diff
@@ -24,15 +24,26 @@ class HistoryController: NSObject {
   }
 
   private func read() {
-    history = (NSKeyedUnarchiver.unarchiveObject(withFile: plistURL.path) as? [PlaybackHistory]) ?? []
+    // Avoid logging a scary error if the file does not exist.
+    guard FileManager.default.fileExists(atPath: plistURL.path) else { return }
+    do {
+      let data = try Data(contentsOf: plistURL)
+      let object = try NSKeyedUnarchiver.unarchivedObject(ofClasses: [NSArray.self, PlaybackHistory.self],
+                                                          from: data)
+      history = object as? [PlaybackHistory] ?? []
```

Comment:
I coded it that way with error logging and then simplified it and put it back to the way the code originally was. I prefer simplifying further to the code below. The other error handling needs to be there as you can never assume an external file hasn't been messed with. But the call to `unarchivedObject` will throw an error if the file contains any classes other than an array or `PlaybackHistory` objects. A developer would have to change `save` to save something like nested arrays, or just a single  `PlaybackHistory` object to cause `unarchivedObject` to be successful and the cast to fail. I don't see the need to cover that case, but I'm happy to add in the check and logging if you prefer an more cautious approach. Thoughts?
```suggestion
      history = try NSKeyedUnarchiver.unarchivedObject(ofClasses: [NSArray.self, PlaybackHistory.self],
                                                       from: data) as! [PlaybackHistory]
```

Reply:
I think we don't need extra handling here.